docs/RFCS/20220810_tenant_capabilities.md

```diff
@@ -0,0 +1,466 @@
+and we grant `SET_SYSTEM_CLUSTER_SETTING` to tenant with ID 2,
+and an operator runs `SET CLUSTER SETTING` on tenant 2, we
+can't keep the current code that writes to tenant 2's `system.settings` table.
+
+Instead, we must **create a new KV/SQL networked API** which channels
+the setting write from tenant 2, through the KV layer, to become
+a write in tenant 1's keyspace.
+
+The new logic would then be:
+
+1. check if the current tenant currently has the `SET_CLUSTER_SETTING` capability
+   for the target SystemOnly setting. If not, report an error to the user.
+2. send a `SettingChange` request (NEW) to the KV Batch interface, as before.
+3. on the KV side, `SettingChange` is refused if the origin tenant
+   does not have the SET_CLUSTER_SETTING capability for the target setting.
+4. on the KV side, the `SettingChange` request is then translated to
+   a `SET CLUSTER SETTING` SQL statement executed in the system/admin tenant.
+
+### Administrating capabilities
+
+We could build additional SQL syntax to set capabilities via an
+additional parameter to the SQL built-in function
+`crdb_internal.create_tenant()` (via a new overload), and/or a new
+`crdb_internal.set_tenant_capabilities()` built-in to edit existing
+tenants.
+
+However note this is not strictly required, since an operator
+can also edit the `system.tenants` table directly and the `info`
+column via the `crdb_internal.pb_to_json` and `json_to_pb` built-ins.
+
+Note that the permission to change capabilities for tenants will *not*
+be itself a capability. It will be solely linked to the current SQL
+user's ability to write to the *current tenant's* `system.tenants`
+table. See next section.
+
+### Deciding which tenant is responsible for hosting cluster configuration
+
+Previously we've assumed that there's just one tenant containing the
+`system.tenants`, `system.tenant_usage` and `system.tenant_settings`
+tables, and that this tenant was always the one with ID 1, and we could
+use that equivalence to look up these system tables from anywhere.
+
+The same applies for the cluster-wide setting values (the
+authoritative tenant's `system.settings`), span configs
+(`system.span_configurations`), etc.
+
+In a world where the tenant ID is not any more the factor that
+determines whether a tenant can "administrate the cluster", we need a
+new mechanism to point a cluster to *which tenant is currently
+authoritative for the cluster-wide system tables*.
+
+We will achieve this by introducing a new key in the system config
+(alongside e.g. liveness and the range ID gen), `SystemTenantID`. Its
+initial value will be 1.
+
+Whenever we need to look up cluster-wide configuration in the KV
+layer, we will not any more use the compile-time constant 1 as the
+system tenant ID, and instead refer to that run-time configurable key.
+
+This configuration variable needs to be a system config key, and not a
+cluster setting, because we need its value before we can set up the
+cluster settings subsystem.
```

Comment:
What is a "system config key" and what do you mean "before we can set up the cluster settings subsytem?